app/Actions/Photo/Setter.php

```diff
@@ -17,7 +17,8 @@ class Setter
 
 	public function do(string $photoID, string $value): bool
 	{
-		$photo = Photo::findOrFail($photoID);
+		/** @var Photo $photo */
+		$photo = Photo::query()->findOrFail($photoID);
```

Comment:
It does nothing.
It is just an alternative way to write the same query.
The big difference is by doing `query()` you get auto-completion (if I remember correctly).

Reply:
Yes. You get auto-completion and it is also safer. Eloquent does a lot of "call magic" through the magic PHP function `__call` which IMHO is a really bad design decision. (There are also some rants about that out there in the web.)

Basically, the following happens. The method `findOrFails` is not defined by the class `Photo` or one of its parent classes. As the method is undefined, the magic function `__call` is invoked which is overwritten by Eloquent. Eloquent implements the method `__call` such that it checks a list of several other classes, looks for a class which defines the method `findOrFails` and then calls `findOrFails` on an instance of this class. In this particular case, it happens that this "other object" is an instance of the Eloquent class `QueryBuilder` and an instance of this class is returned by `query()`.

The problem about this approach is that if the programmer does a minor misspelling it might happen that a completely different object (which happens to have implemented the misspelled method) is called and you end up with very hard to track errors.

So I am convinced it is part of a defensive programming strategy to be explicit about what you want. I want to call the method `findOrFail` on the `QueryBuilder` and if (for some reason) I type `findsOrFail` I want to get a PHP error which tells me that `QueryBuilder` does not define the method and which directly points to the line of source code where I did the error. I don't want to get an obscure error message which points deeply into some Eloquent class, because the Eloquent "magic" has fallen through up to this point.

For the same reason I am so fussy about type hints.

On top, this call magic interacts with PHP xdebug in a very unpleasant way. If I try to find bugs, I really like to use an interactive debugger which allows me to interactively step across statements and also step in/return from methods. However, if the method to be called is resolved programmatically, this does not really work.

Moreover it is faster and safes some CPU cycles to directly call the method :-)